### PR TITLE
fix: Add missing variables in env file in config

### PR DESCRIPTION
**Pull request by sharat87**
The env config API currently only _changes_ values that are already defined in the env file. It is not capable of adding anything to the file.

This PR adds the capability to do so. However, since we don't want to let the client add just any variable, we've switched from a black list to a white list of env variables that can be managed by this API. If the requested variable is present in the whitelist, we add it, if its missing in the env file.

@abhvsn, I recall you suggested we do a white list earlier, here we are! 😉

### CODE REVIEW

**Comment by sharat87** [comment]
Depends on #7043 to be merged first.

File: app/server/appsmith-server/src/test/java/com/appsmith/server/solutions/EnvManagerTest.java [hunk 1]
```diff
@@ -8,113 +9,147 @@
...
+    public void disallowedVariable() {
+        final String content = "APPSMITH_MONGODB_URI=first value\nDISALLOWED_NASTY_STUFF=\"quoted value\"\n\nAPPSMITH_INSTANCE_NAME=third value";
+
+        assertThatThrownBy(() -> EnvManager.transformEnvContent(
+                content,
+                Map.of(
+                        "APPSMITH_MONGODB_URI", "new first value",
+                        "DISALLOWED_NASTY_STUFF", "new second value"
+                )
+        )).isInstanceOf(AppsmithException.class);
```
Comment (abhvsn) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/solutions/EnvManagerTest.java:132]: Can we compare the actual error message here `AppsmithError.UNAUTHORIZED_ACCESS`?

**Review by abhvsn — COMMENTED** [review]
LGTM

**Review by abhvsn — DISMISSED** [review]

**Review by abhvsn — DISMISSED** [review]

**Comment by sharat87** [comment]
@abhvsn, resolved merge conflicts. Can you review again please?

**Review by abhvsn — APPROVED** [review]

**Pull request merged**